drivers/clock_mgmt/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+zephyr_linker_sources(RWDATA clock_mgmt_callbacks.ld)
+zephyr_library_sources(clock_callbacks.c)
+
+# Helper macro to set clock management driver
+macro(zephyr_clock_mgmt_driver_ifdef toggle driver)
+  if(${toggle})
+    set(clock_mgmt_driver ${CMAKE_CURRENT_SOURCE_DIR}/${driver})
```

Comment:
Bit confused by this, it seems it's only possible to have 1 clock mgmt driver?

Reply:
Yes- I should have made that more clear in the RFC body. Like pin control, this subsystem is designed to be single instance. The idea is that an soc clock control driver is capable of handling all SOC clocking configurations and clock subsystems

Reply:
How will this work for complex systems involving external clock generators? Or FPGAs with multiple clocking subsystems?

Reply:
Both good questions. For an FPGA with multiple clocking subsystems, how would this differ from an SOC? My understanding is that once a given bitstream is loaded to an FPGA, you could then load a firmware that handled the clock tree implemented within that bitstream.

The external clock generator issue is a bit more complicated, and a good case to motivate multiple clock devices. The core reason I wanted to avoid this (as mentioned above) was that I want to use the phandle references within a node setpoint to refer to clock nodes, not the clock device. I guess a general question would be does this approach make sense? I like the simplicity it lends to most configurations, but it comes with tradeoffs in terms of new tooling (as I hope this PR makes clear)

Since we are already changing the devicetree tooling for this PR, perhaps I could add code to find the "clock controller" device for a given clock node (which should be above the clock node in the device tree hierarchy), and generate a reference to that node?

Reply:
> How will this work for complex systems involving external clock generators? Or FPGAs with multiple clocking subsystems?

My interpretation of this RFC is that it is an SOC clock management framework. What would need to be addressed regarding external clock sources in this system? Wouldn't the external devices still be configured by some driver?